client/client.go

```diff
@@ -349,11 +349,14 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 
 	select {
 	case <-ctx.Done():
-		err = resp.Body.Close()
-		<-done
-		if err == nil {
-			err = ctx.Err()
+		if err = resp.Body.Close(); err != nil {
```

Comment:
> I do not think you understand what I was talking about. There is a done chan without buffer. If you break out at line 353, then the other go routine will block at line 347.

You are right. The read goroutine will block forever. I didn't see that before. :(

> When does body.Close return an error?

In normal workflow for context timeout, body.Close doesn't return error here. It may return error in some special case, e.g., HTTP has the trailer but it cannot read it out.

The reason that it checks error here and breaks early is to ensure that the code doesn't replace it with `ctx.Err` blindly.